### ISSUE TITLE
Menus do not react when mouse hovers, unlike all other clickable components

### ISSUE THREAD

**Issue by Chrriis** [JFormDesigner/FlatLaf#49]
Under FlatLaf, components change style on mouse hover: buttons get a different border, tabs get a gray background, etc. It makes the application feel "alive" too.
Menus do not react and feel "dead". I think they should get a gray background, like tabs.

I also wonder if an "tab" style would be appropriate, like toggle buttons.

Simple test case of a menu bar:
```Java
	public static void main(String[] args) throws Exception {
		UIManager.setLookAndFeel(new FlatLightLaf());
		JFrame frame = new JFrame();
		frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
		JMenuBar menuBar = new JMenuBar();
		JMenu menu1 = new JMenu("Menu 1");
		menu1.setMnemonic('m');
		JMenuItem subMenu1_1 = new JMenuItem("Sub Menu 1.1");
		subMenu1_1.setMnemonic('s');
		menu1.add(subMenu1_1);
		menuBar.add(menu1);
		JMenu menu2 = new JMenu("Menu 2");
		menu2.setMnemonic('n');
		JMenuItem subMenu2_1 = new JMenuItem("Sub Menu 2.1");
		subMenu2_1.setMnemonic('u');
		menu2.add(subMenu2_1);
		menuBar.add(menu2);
		frame.setJMenuBar(menuBar);
		frame.setContentPane(new JScrollPane(new JTextArea()));
		frame.setSize(400, 300);
		frame.setVisible(true);
	}
```

**Comment by DevCharly** [comment]
Yes, that's missing. 

Now implemented in master branch 😃

**Comment by Chrriis** [comment]
Yes, much better! Thanks a lot!

What do you think of the suggestion of supporting a "tab" type for menus, similar to toggle buttons? Is it valid? Should I open a separate issue?

**Comment by DevCharly** [comment]
Something like this?
![image](https://user-images.githubusercontent.com/5604048/72905733-46476f80-3d31-11ea-9f16-59737598c877.png)

I think this may be confusing because the "tab" style is used to switch content.

**Comment by DevCharly** [comment]
hover in menu bar is implemented in 0.26

**Comment by Chrriis** [comment]
> I think this may be confusing because the "tab" style is used to switch content.

Well, you are right if the hover behaved like a tab. Rather, I had the following in mind:
![MenusWithTabStyle](https://user-images.githubusercontent.com/1174202/72917514-469d3600-3d44-11ea-88fe-5e4e65e4dbf9.png)

So:
- hover would be like now.
- click shows selection like a tab. In effect, it is selected.

**Comment by DevCharly** [comment]
But then selected menu looks like a tab...
What should then selected menu items look like?

**Comment by Chrriis** [comment]
> But then selected menu looks like a tab...

True, but only when the sub-menu is showing: you know what got activated. Currently, selection behaves like a tree or a list (is that better?):
![MenusWithTabStyle2Original](https://user-images.githubusercontent.com/1174202/72929468-2c6e5280-3d5a-11ea-9130-1385fd7fefd0.png)

My initial reaction was that with a modern L&F like FlatLaf, something better could be found in this area than a Windows 95 kind of selection 🙂 
The other candidate I could think of is the underline selection indicator. I am sure I saw it on some websites for some menus but I cannot get my hands on one of them... Nevertheless, the semantics are not far off: selection reveals a component.

This is just brainstorming, and your question is legitimate: how would sub-menus behave? Well, they could remain the same, or they could have a similar style. Here is an attempt:
![MenusWithTabStyle2Possibility](https://user-images.githubusercontent.com/1174202/72929833-e5349180-3d5a-11ea-9b8a-b7e6767e3cff.png)

Hope this helps!

**Comment by Chrriis** [comment]
I found some websites with underline menus. Here is one example: https://www.zalando.fr
You just hover to show the popup (no click), but it is definitely not a tab.

**Comment by DevCharly** [comment]
@Chrriis while working on new menu item layout to fix #3, I've also implemented your idea.
Looks nice 👍 

![image](https://user-images.githubusercontent.com/5604048/80596126-23121780-8a26-11ea-9a7c-023a910ba046.png)

Can be enabled with:

~~~java
UIManager.put( "MenuItem.selectionType", "underline" );
~~~
